mlflow/environment_variables.py

```diff
@@ -789,3 +789,11 @@ def get(self):
 MLFLOW_PRINT_MODEL_URLS_ON_CREATION = _BooleanEnvironmentVariable(
     "MLFLOW_PRINT_MODEL_URLS_ON_CREATION", True
 )
+
+#: Maximum number of threads to use when downloading traces during search operations.
+#: If unspecified the number of threads (at least 32) is determined by the available
+#: number of CPU cores.
+#: (default: ``None``)
+MLFLOW_SEARCH_TRACES_MAX_THREADS = _EnvironmentVariable(
+    "MLFLOW_SEARCH_TRACES_MAX_THREADS", int, None
```

Comment:
```suggestion
    "MLFLOW_SEARCH_TRACES_MAX_THREADS", int, max(32, (os.cpu_count() or 1) * 4)
```

is there a reason that we don't set the default here?

Reply:
As discussed offline, we can, in fact, document this clearly, so I've made this change